Actions/AL-Go-TestRepoHelper.ps1

```diff
@@ -64,12 +64,14 @@ function Test-ALGoRepository {
         [string] $baseFolder
     )
     
+    Write-Host "BaseFolder: $baseFolder"
+
     # Test .json files are formatted correctly
     Get-ChildItem -Path $baseFolder -Filter '*.json' -Recurse | ForEach-Object {
-        if ($_.FullName -like '*\.AL-Go\Settings.json') {
+        if ($_.FullName -like "*$([System.IO.Path]::DirectorySeparatorChar).AL-Go$([System.IO.Path]::DirectorySeparatorChar)settings.json") {
             Test-Json -jsonFile $_.FullName -baseFolder $baseFolder
         }
-        elseif ($_.FullName -like '*\.github\*Settings.json') {
+        elseif ($_.FullName -like "*$([System.IO.Path]::DirectorySeparatorChar).github$([System.IO.Path]::DirectorySeparatorChar)*ettings.json") {
```

Comment:
```suggestion
        elseif ($_.FullName -like "*$([System.IO.Path]::DirectorySeparatorChar).github$([System.IO.Path]::DirectorySeparatorChar)settings.json") {
```

Maybe? Hopefully?

Reply:
No, This is reading all settings files in .github folder (and tests validity):
![image](https://user-images.githubusercontent.com/10775043/207308795-0e681f68-53b1-40e3-bc20-87afd8263d2a.png)

Reply:
Wait, I do not get it... why isn't `*settings.json"` matching all settings files?

Reply:
Linux has case sensitive file names - and we have:
.AL-Go\settings.json (project settings)
.github\AL-Go-Settings.json (repo settings)
.github\CICD.settings.json (workflow settings)

The code after removing the directory separators is:

```
    # Test .json files are formatted correctly
    Get-ChildItem -Path $baseFolder -Filter '*.json' -Recurse | ForEach-Object {
        if ($_.FullName -like (Join-Path "*/.AL-Go" "settings.json")) {
            Test-Json -jsonFile $_.FullName -baseFolder $baseFolder
        }
        elseif ($_.FullName -like (Join-Path "*/.github" "*ettings.json")) {
            Test-Json -jsonFile $_.FullName -baseFolder $baseFolder -repo:($_.BaseName -eq "AL-Go-Settings")
        }
    }
```
I guess this would work as well:

```
Get-ChildItem -Path $baseFolder -Filter '*ettings.json' -Recurse -File | Where-Object { $_.Directory.Name -in @(".AL-Go", '.github') } | ForEach-Object {
    Test-Json -jsonFile $_.FullName -baseFolder $baseFolder -repo:($_.BaseName -eq "AL-Go-Settings")
}
```

Reply:
If I change the code to this, then it will also test files in .AL-Go called oldsettings.json - which could cause a build failure somewhere.
(I actually had a file like that in a local project)

Reply:
changed the code to a compromise, which shouldn't give any problems in existing repos:

```
    Get-ChildItem -Path $baseFolder -Filter '*.json' -Recurse | ForEach-Object {
        if ($_.DirectoryName -eq '.AL-Go' -and $_.BaseName -eq 'settings') {
            Test-Json -jsonFile $_.FullName -baseFolder $baseFolder
        }
        elseif ($_.DirectoryName -eq '.github' -and $_.BaseName -like '*ettings') {
            Test-Json -jsonFile $_.FullName -baseFolder $baseFolder -repo:($_.BaseName -eq 'AL-Go-Settings')
        }
    }
```